src/OrchardCore/OrchardCore.Users.Core/Services/UserStore.cs

```diff
@@ -524,5 +534,17 @@ public async Task<IList<IUser>> GetUsersForClaimAsync(Claim claim, CancellationT
         }
 
         #endregion IUserClaimStore<IUser>
+
+        #region IQueryableuserStore
+        public IQueryable<IUser> Users => GetUsersAsync().GetAwaiter().GetResult().Roles.AsQueryable();
+        #endregion
+
+        private async Task<UsersDocument> GetUsersAsync()
+        {
+            return await _scopedDistributedCache.GetOrSetAsync(() =>
```

Comment:
don't cache that, it's already bad to call it as it's an unbounded query. what if there are 1 million or more users?

Reply:
What about `Roles`? I saw similar behavior

Reply:
you may have a million of users but not a million of roles

Reply:
In the related issue i said `Maybe we should add this`, thinking that anyway somebody can already do an unbounded query through the session, but yes maybe better to not add it explicitly to the interface.

Reply:
Ok, I just update the PR and I will need to test the admin UI to make sure everything work as expected